service-workers/service-worker/resources/request-end-to-end-worker.js

```diff
@@ -1,9 +1,20 @@
 var port = undefined;
+// Create a then-able object that is never resolved.
+function createPending() {
+  return { then: createPending };
+}
 
 onmessage = function(e) {
   var message = e.data;
   if (typeof message === 'object' && 'port' in message) {
     port = message.port;
+
+    port.postMessage('received port');
+    // The ServiceWorker which handles the "message" event must persist long
+    // enough to handle the subsequent "fetch" event. To promote test
+    // simplicity, the worker prevents its own termination indefinitely via a
+    // then-able that is never resolved.
+    e.waitUntil(createPending());
```

Comment:
So, this changed caused a number of other tests to fail in firefox.  Consider:

1. This test uses a scope of 'resources/blank.html' which many other tests also use.
2. It keeps the service worker thread alive until the browser force kills it.
3. We have agreement to delay marking a SW redundant until it goes idle in https://github.com/w3c/ServiceWorker/issues/916

This means this change keeps the SW registration from this test alive until the force kill timeout is triggered (5 minutes on FF) and blocks any other registration from being made on the same scope.

To work around this for now I'm moving this test to a more unique scope.  In the long run, though, I think this test should change to use a promise that resolves when the test explicitly sends a "done" message or something.  There is no need to keep the SW alive indefinitely.